service/storage_service.cc

```diff
@@ -1608,6 +1608,50 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
     assert(_group0);
     co_await _group0->finish_setup_after_join(*this, _qp, _migration_manager.local(), _raft_experimental_topology);
     co_await _cdc_gens.local().after_join(std::move(cdc_gen_id));
+
```

Comment:
> Add a fiber which is spawned only in legacy mode that will wait until:
>
> - The schema-on-raft upgrade finishes,
> - The SUPPORTS_CONSISTENT_CLUSTER_MANAGEMENT feature is enabled,
> - The upgrade is triggered by the user.

Do we need a fiber to track all of this or each one has some kind of event that can be used to trigger the raf monitor fiber creation?

Reply:
Of those three, I think that only `supports_consistent_topology_changes` allows that through `gms::feature::when_enabled`. Others (`wait_until_group0_upgraded` and waiting for topology state machine to enter the right state) requires a fiber that waits for the right events.